### PR TITLE
fix(opencode): close shell spill stream via scoped finalizer

### LINKED ISSUE #1051: [Bug] Shell spill tempfile stream not closed on interrupt/defect (use scoped finalizer)
## What happened?

In the shell tool's `run()` (`packages/opencode/src/tool/shell.ts`), when command output exceeds the in-memory limit it spills to a tempfile via a write stream (`sink`, created at `shell.ts:529`). That stream is closed only at the end of the normal path (`shell.ts:606-615`), which runs after the scoped execution block (`shell.ts:492-580`) returns.

If the scoped block is interrupted or dies (defect) between sink creation and that final close — note `shell.ts:580` does `Effect.orDie`, so a failure becomes a defect that unwinds past the close — the write stream is never closed. That leaks a file handle and can drop the unflushed tail of the spill file. Normal exit, abort, and timeout paths all reach the close; only the interrupt/defect path leaks.

Fix: register the stream close as a scoped finalizer (`Effect.acquireRelease` / `addFinalizer`) so it always runs on scope exit regardless of how the scope ends. This also matches current upstream, which already closes the spill stream via a scoped finalizer.

## Which area seems affected?

Model harness, prompts, tools, or session mechanics.

## How much does this affect you?

Minor issue or polish (latent resource cleanup; no security or user-file impact).

## Steps to reproduce

Hard to trigger deterministically. Requires a command whose output spills to a tempfile, with the `run()` effect interrupted or hitting a defect mid-stream before the normal close runs.

Code path: `shell.ts:529` (create), `shell.ts:606-615` (close), `shell.ts:492-580` (scoped block + `Effect.orDie`).

## What did you expect to happen?

The spill-file write stream should always be closed and flushed on scope exit, regardless of whether the scope ends by exit, abort, timeout, interrupt, or defect.

## PawWork version

dev (current `dev` branch).

## OS version

All (macOS and Windows).

## Can you reproduce it again?

Only once so far (latent / error-path only).

## Diagnostics

Found during the #1038 shell-tool review. Low-risk, self-contained fix; convergent with upstream.

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

Fixes #1051. The shell tool's spill write stream (`sink`) was closed only after the `Effect.scoped(...).pipe(Effect.orDie)` block returned. On the interrupt/defect path — `orDie` turns a failure into a defect that unwinds past the close — the stream was never closed, leaking a file handle and dropping the unflushed spill tail. Normal exit, abort, and timeout already reached the close; only interrupt/defect leaked.

## Change

- Register the spill-stream close as a scoped finalizer (`Effect.addFinalizer`) so it always runs on scope exit regardless of how the scope ends, matching upstream.
- Register it before the consumer `Effect.forkScoped` so the consumer fiber is interrupted first (finalizers run LIFO), leaving no write racing the close.
- Remove the now-redundant post-scope close block.

## Touched files

- `packages/opencode/src/tool/shell.ts` (only)

## Verification

- `bun run typecheck` (opencode): pass
- `bun test test/tool/shell.test.ts test/tool/truncation.test.ts`: 64 pass / 0 fail
- `/codex review` (xhigh; focus on LIFO ordering, flush behavior, edge cases): no P1/P2 findings
- Normal / abort / timeout paths are behaviorally unchanged (the scoped block awaits finalizers before returning); only the interrupt/defect path now closes and flushes the stream. A deterministic interrupt repro is not feasible (the issue notes the same); correctness is by construction.

## Risk

Low. Single-file, self-contained, convergent with upstream. No public API or output-format change.

Closes #1051

### CODE REVIEW

File: packages/opencode/src/tool/shell.ts [hunk 1]
```diff
@@ -493,6 +493,21 @@ export const ShellTool = Tool.define(
...
+          yield* Effect.addFinalizer(() =>
+            Effect.promise(
+              () =>
+                new Promise<void>((resolve) => {
+                  if (!sink) return resolve()
+                  sink.end(() => resolve())
+                  sink.on("error", () => resolve())
+                }),
+            ),
+          )
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/tool/shell.ts:509]: Verified false positive for `fs.WriteStream`. Node emits the stream `error` event asynchronously (on a later tick / during async I/O), not synchronously from within `end()`. Both `end(cb)` and `on("error")` are registered in the same synchronous Promise-executor tick, so any error surfaces after both handlers are in place — their relative order is immaterial here. This also preserves the close ordering of the post-scope block this finalizer replaces. Resolving.
